### DISCUSSION TITLE
"Redeploy with existing Build Cache." and environment variables

### DISCUSSION THREAD

**Discussion by ksb86** [vercel/next.js#40120] [Help]
When redeploying through the vercel UI, if the "Redeploy with existing Build Cache." checkbox is checked, is the new build supposed to pick up environment variable changes since the last deploy?

**Comment by gaston-flores** [comment]
Yes, Vercel always provides the latest environment variables on every deploy. Build cache is used to store things like `node_modules` and framework specific files. [You can see what the cache stores in the documentation.](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#what-is-cached)
